docs/spec_draft.md

```diff
@@ -1808,6 +1809,64 @@ where `i[d] = j[permutation[d]]`.
 
 [Back to Ops](#index-of-ops)
 
+## stablehlo.while
+
+### Semantics
+Produces the output from executing `body` function 0 or more times till the
+`cond` function outputs `true`. More formally, `body` function is executed once
+every time the `cond` function returns `true` and all execution stops once
+`cond` function returns `false`. For the first iteration both `cond` and `body`
+arguments are initialzed with the vaules of arguments of the `while` operation,
+for subsequent iterations they are fed with the output of `body` from the
+previous itertaion.
```

Comment:
When reframed as pseudocode, this description suggests the question of what happens if the loop ends up being infinite. Do we allow that? Do we disallow that? Do we call this undefined behavior?

Reply:
the interpreter suggests that, It is allowed and results in undefined behaviour